src/main/java/com/flightstats/hub/webhook/WebhookService.java

```diff
@@ -37,27 +46,33 @@ public WebhookService(@Named("Webhook") Dao<Webhook> webhookDao,
                           WebhookManager webhookManager,
                           LastContentPath lastContentPath,
                           ChannelService channelService,
+                          ContentRetriever contentRetriever,
+                          LocalWebhookManager localWebhookManager,
                           WebhookProperties webhookProperties) {
         this.webhookDao = webhookDao;
         this.webhookValidator = webhookValidator;
         this.webhookManager = webhookManager;
         this.lastContentPath = lastContentPath;
         this.channelService = channelService;
+        this.contentRetriever = contentRetriever;
+        this.localWebhookManager = localWebhookManager;
         this.webhookProperties = webhookProperties;
     }
 
     public Optional<Webhook> upsert(Webhook webhook) {
-        logger.info("incoming webhook {} ", webhook);
+        log.info("incoming webhook {} ", webhook);
         webhook = webhook.withDefaults(webhookProperties.getCallbackTimeoutDefaultInSec());
         webhookValidator.validate(webhook);
-        Optional<Webhook> preExisting = get(webhook.getName());
+        final Optional<Webhook> preExisting = get(webhook.getName());
         if (webhook.isTagPrototype()) {
-            return upsertTagWebhook(webhook, preExisting);
+            this.upsertTagWebhook(webhook);
+            return preExisting;
         }
         if (preExisting.isPresent()) {
-            Webhook existing = preExisting.get();
-            ContentPath newStartingKey = webhook.getStartingKey();
-            if (newStartingKey != null && newStartingKey.getTime().compareTo(new DateTime(1980, 1, 1, 1, 1)) >= 0
+            final Webhook existing = preExisting.get();
+            final ContentPath newStartingKey = webhook.getStartingKey();
+            if (newStartingKey != null
```

Comment:
As discussed,  this is out of scope of ticket. Since this particular ticket is about removing static injection and  it touches most parts of the hub, I would avoid any other refactor here. This will help in debugging any issues that might crop up. Any other code refactor should be taken up separately and should be refined to particular functional area.